code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java

```diff
@@ -42,67 +38,102 @@
+			// Mock MobileCore.registerExtension()
+			ArgumentCaptor<Class> extensionClassCaptor = ArgumentCaptor.forClass(Class.class);
+			ArgumentCaptor<ExtensionErrorCallback> callbackCaptor = ArgumentCaptor.forClass(
+				ExtensionErrorCallback.class
+			);
+			mobileCoreMockedStatic
+				.when(() -> MobileCore.registerExtension(extensionClassCaptor.capture(), callbackCaptor.capture()))
+				.thenReturn(true);
+			// Call deprecated registerExtension() API
+			EdgeBridge.registerExtension();
+			// Verify: happy path - extension registered is EdgeBridgeExtension
+			assertNotNull(callbackCaptor.getValue());
+			assertEquals(EdgeBridgeExtension.class, extensionClassCaptor.getValue());
+			// Verify: captured error callback does not throw NPE when passed null ExtensionError
+			callbackCaptor.getValue().error(null);
+		}
+	}
+
 	@Test
 	public void test_listenerRegistration() {
-		final ArgumentCaptor<ExtensionErrorCallback> callbackCaptor = ArgumentCaptor.forClass(
-			ExtensionErrorCallback.class
+		extension.onRegistered();
+		ArgumentCaptor<String> eventTypeCaptor = ArgumentCaptor.forClass(String.class);
+		ArgumentCaptor<String> eventSourceCaptor = ArgumentCaptor.forClass(String.class);
+		ArgumentCaptor<ExtensionEventListener> extensionEventListenerArgumentCaptor = ArgumentCaptor.forClass(
+			ExtensionEventListener.class
 		);
-
+		// Verify: 2 event listeners registered; capture values from registration
 		verify(mockExtensionApi, times(2))
-			.registerEventListener(anyString(), anyString(), any(Class.class), any(ExtensionErrorCallback.class));
-
-		verify(mockExtensionApi, times(1))
 			.registerEventListener(
-				eq(EdgeBridgeTestConstants.EventType.GENERIC_TRACK),
-				eq(EdgeBridgeTestConstants.EventSource.REQUEST_CONTENT),
-				eq(ListenerGenericTrackRequestContent.class),
-				callbackCaptor.capture()
+				eventTypeCaptor.capture(),
+				eventSourceCaptor.capture(),
+				extensionEventListenerArgumentCaptor.capture()
 			);
-		assertNotNull(
-			"The extension callback for Generic Track listener should not be null",
-			callbackCaptor.getValue()
-		);
 
-		verify(mockExtensionApi, times(1))
-			.registerEventListener(
-				eq(EdgeBridgeTestConstants.EventType.RULES_ENGINE),
-				eq(EdgeBridgeTestConstants.EventSource.RESPONSE_CONTENT),
-				eq(ListenerRulesEngineResponseContent.class),
-				callbackCaptor.capture()
-			);
-		assertNotNull("The extension callback for Rules Engine listener should not be null", callbackCaptor.getValue());
+		// Extract captured values into lists
+		List<String> eventTypes = eventTypeCaptor.getAllValues();
+		List<String> eventSources = eventSourceCaptor.getAllValues();
+		List<ExtensionEventListener> extensionEventListenerList = extensionEventListenerArgumentCaptor.getAllValues();
+
+		// Verify: 1st Edge Bridge event listener
+		assertEquals(EdgeBridgeTestConstants.EventType.GENERIC_TRACK, eventTypes.get(0));
```

Comment:
for all these asserts can we use the core eventype/source instead?

Reply:
I think for your tests you can assume core value will not change since they are publicly shared constants. In this case catching unexpectedly changed values should be part of core tests